test/test_ir.py

```diff
@@ -1,16 +1,81 @@
 from __future__ import annotations
 
 import math
+from collections.abc import Sequence
+from typing import Any
 
 import numpy as np
 import pytest
+from numpy.typing import ArrayLike
 
 from opensquirrel.common import ATOL
-from opensquirrel.ir import BlochSphereRotation, ControlledGate, MatrixGate, Measure, Qubit
+from opensquirrel.ir import Axis, BlochSphereRotation, ControlledGate, Expression, MatrixGate, Measure, Qubit
+
+
+class TestAxis:
+
+    @pytest.fixture(name="axis")
+    def axis_fixture(self) -> Axis:
+        return Axis(1, 0, 0)
+
+    @pytest.mark.parametrize("expected_class", [Sequence, Expression])
+    def test_inheritance(self, axis: Axis, expected_class: type[Any]) -> None:
+        assert isinstance(axis, expected_class)
+
+    def test_axis_getter(self, axis: Axis) -> None:
+        np.testing.assert_array_equal(axis.axis, [1, 0, 0])
+
+    @pytest.mark.parametrize(
+        "new_axis, expected_axis", [([0, 0, 1], [0, 0, 1]), ([0, 3, 4], [0, 3 / 5, 4 / 5]), (Axis(0, 1, 0), [0, 1, 0])]
+    )
+    def test_axis_setter_no_error(self, axis: Axis, new_axis: ArrayLike, expected_axis: ArrayLike) -> None:
+        axis.axis = new_axis
+        np.testing.assert_array_equal(axis.axis, expected_axis)
+
+    @pytest.mark.parametrize(
+        "erroneous_axis, expected_error, expected_error_message",
+        [
+            (Qubit(1), TypeError, "Axis requires an ArrayLike"),
+            ([0, [3], [2]], TypeError, "Axis requires an ArrayLike"),
+            (0, ValueError, "Axis requires an ArrayLike of length 3, but received an ArrayLike of length 1."),
+            (
+                [1, 2, 3, 4],
+                ValueError,
+                "Axis requires an ArrayLike of length 3, but received an ArrayLike of length 4.",
+            ),
+        ],
+    )
+    def test_axis_setter_with_error(
+        self, axis: Axis, erroneous_axis: Any, expected_error: type[Exception], expected_error_message: str
+    ) -> None:
+        with pytest.raises(expected_error, match=expected_error_message):
+            axis.axis = erroneous_axis
+
+    def test_get_item(self, axis: Axis) -> None:
+        assert axis[0] == 1
+        assert axis[1] == 0
+        assert axis[2] == 0
+
+    def test_len(self, axis: Axis) -> None:
+        assert len(axis) == 3
+
+    def test_repr(self, axis: Axis) -> None:
+        assert repr(axis) == "Axis[1. 0. 0.]"
```

Comment:
Hm... why was I expecting `Axis[1., 0., 0.]`? But I have to say I prefer this repr instead, yes.

Reply:
This is the default representation of a `numpy` array, it was already used somewhere else, so I thought it would be nice to use it here as well.

Reply:
Fair enough!